### ISSUE TITLE
Allow users to manually specify a project thumbnail

### ISSUE THREAD

**Issue by benjiwheeler** [LLK/scratch-www#4092]
It would be good to somehow allow users to manually specify a project thumbnail. That could work several different ways:

1. treat them like user avatars or studio avatars -- users can upload an image. This would probably lead to lots of uploading, which our servers are not great at handling.
2. make a "take a snapshot" button of some sort, that uses the current stage to make a thumbnail
3. something else? or a combination?

**Comment by BoomerScratch** [comment]
If this would get implemented, there must also be an option when you report a project.
![](http://u.cubeupload.com/nobody555/thumbnailreport.png)

**Comment by R4356th** [comment]
@BoomerScratch, doesn't Inappropriate Images already cover them? :stuck_out_tongue_winking_eye:

**Comment by apple502j** [comment]
Hmm, does it cause increase in traffic so much? It is saved every time you save the project. Also, if we keep automated uploading it may override the manual one.

**Comment by BoomerScratch** [comment]
@R4356th I guess that is for the project... While on the project page, you can't see the thumbnail, unless it is in the project.

**Comment by R4356th** [comment]
> @R4356th I guess that is for the project... While on the project page, you can't see the thumbnail, unless it is in the project.

I do not understand. A thumbnail is still an image, you know?

**Comment by Explosion-Scratch** [comment]
Hmm, while this would be better the probpem is existing thumbnails. There are many methods to creating them and it would be bad if they all got deleted. What about something close to the backdrop design right below it called "Thumbnail"? That way if it was blank it would use the current system (because the thumbnail would be transparent and reveal the coded thumbnail) This would bassically be just the paint editor and make the thumbnail.

**Comment by BoomerScratch** [comment]
@R4356th Yes, but the ST might not notice it.

**Comment by Explosion-Scratch** [comment]
@BoomerScratch, are you saying that if it dosen't appear on the project project page that it might cause moderation issues? The ST can't see the current thumbnails for a few ways of making them when they aren't on the project page.

**Comment by Kenny2github** [comment]
Opinions:
- If/when this change is made, existing project-snapshot thumbnails get converted into manually set thumbnails in one fell swoop
- Users can manually set thumbnails (with certain restrictions on image filesize and aspect ratio), *with the option to take a snapshot of the Stage*
- A new report option is added, "Inappropriate Thumbnail", like @BoomerScratch demonstrated

Responses:
> This would probably lead to lots of uploading, which our servers are not great at handling.

Currently, a thumbnail is saved (and thus uploaded) every time the project is saved anyway, so this would actually reduce load.

> doesn't Inappropriate Images already cover them?

Why have a separate item for "Too Violent or Scary", then, when "Inappropriate Images" could cover that too? It's also useful to have a separate item so that the ST knows where to look immediately, instead of having to figure out whether it's a random image in the project or the thumbnail that caused the report.

**Comment by R4356th** [comment]
> Why have a separate item for "Too Violent or Scary", then, when "Inappropriate Images" could cover that too? It's also useful to have a separate item so that the ST knows where to look immediately, instead of having to figure out whether it's a random image in the project or the thumbnail that caused the report.

I agree with the last part (it would be helpful for the ST...) but not the first part because there might be other inappropriate things in the project which are not covered by Inappropriate Images, e.g. inappropriate language.

**Comment by GrahamSH-LLK** [comment]
**EndlessBlaze commented this on #4126, I just moved it here**
As much as i love the idea of having custom thumbnails the only problem would be, if some people still wanted to use the Manuel thumbnail
I suggest that we keep a manual way to put a thumbnail and also have the automatic option

Maybe when you press share you can choose one of 2 options which are.......

Would you like to use an auto generated image from your project?
Would you like to upload a thumbnail for your project?

**Comment by Explosion-Scratch** [comment]
@GrahamSH-LLK @qucchia made a great mockup of this over of the scratch forums, basically it's a dialog that asks for confirmation before you share but it also has the notes and credits and the instructions of the project there too and you can edit them. The thumbnail part of this is what I really like. It shows the auto-generated thumbnail but underneath has an "upload" button where a user could upload a thumbnail.

**Comment by JeffaloBob** [comment]
> treat them like user avatars or studio avatars -- users can upload an image. This would probably lead to lots of uploading, which our servers are not great at handling.

wouldn't this solution actually lower the amount of uploading? every time you save it makes a thumbnail, right? and like this it'd only be the one time a user chooses to upload a thumbnail. (but I don't really know how that stuff works)


but anyways, this isn't my favourite solution because it kinda prevents thumbnails from being easily made in scratch's custome editor.

something like YouTube's system could be cool, where you have an auto generated thumbnail and you can upload your own, but I'm guessing this would be a lot of uploading and stuff.

my favourite solution is somehow being able to take a snapshot of the screen and have that as a thumbnail, because it's annoying currently if you have a shared project and you want to edit a small part of it, you'd have to show the thumbnail before you leave and I always forget that.

So some kind of way to disable the automatic thumbnail would be nice.

**Comment by GrahamSH-LLK** [comment]
>it kinda prevents thumbnails from being easily made in scratch's custome editor.
> 
I'm assuming you mean the costume editor. If so, couldn't we add a button that sets the current design as the thumbnail?

**Comment by GrahamSH-LLK** [comment]
Another thought. We could set up a system where someone uploads an image or creates a costume like normal, and then they can check a box to "set as thumbnail". If nothing was checked, it would do it automatically. Ie. Something like this. We could automattically disable all blocks on the sprite, and name it "Thumbnail".
![image](https://user-images.githubusercontent.com/64214252/86596067-a50f4700-bf67-11ea-859a-108ec09a65fa.png)

**Comment by Explosion-Scratch** [comment]
Did you guys see the design that qucchia made on the forums? Bassically
it's a share dialog so when you click share there is a dialog that allows
scheduling sharing, changing the notes and credits and uploading a
thumbnail.

**Comment by GrahamSH-LLK** [comment]
No, I can't find it.

**Comment by Ani-Xan** [comment]
> Did you guys see the design that @qucchia made on the forums? Basically, it's a share dialog so when you click share there is a dialog that allows scheduling sharing, changing the notes and credits, and uploading a thumbnail.

@Explosion-Scratch Please give a link to the design by @qucchia you mentioned above.

**Comment by GrahamSH-LLK** [comment]
@benjiwheeler This would be an awesome feature! Are you still thinking about implementing it?
